internal/globalaccounts/globalaccounts.go

```diff
@@ -0,0 +1,228 @@
+	AuthURL        string
+	SubaccountsURL string
+}
+
+func Run(ctx context.Context, cfg Config) {
+	logs := logrus.New()
+	logs.Infof("*** Start at: %s ***", time.Now().Format(time.RFC3339))
+	logs.Infof("is dry run?: %t ", cfg.DryRun)
+
+	svc, db, connection, kcp, err := initAll(ctx, cfg, logs)
+	fatalOnError(err, logs)
+	defer func() {
+		err = connection.Close()
+		if err != nil {
+			logs.Error(err)
+		}
+	}()
+
+	clusterOp, err := clusterOp(ctx, kcp, logs)
+	fatalOnError(err, logs)
+	logs.Println(fmt.Sprintf("No. kymas: %d", len(clusterOp.Items)))
+
+	logic(cfg, svc, db, clusterOp, logs)
+	logs.Infof("*** End at: %s ***", time.Now().Format(time.RFC3339))
+}
+
+func initAll(ctx context.Context, cfg Config, logs *logrus.Logger) (*http.Client, storage.BrokerStorage, *dbr.Connection, client.Client, error) {
+	svcConfig := svcConfig{
+		ClientID:     cfg.AccountServiceID,
+		ClientSecret: cfg.AccountServiceSecret,
+		AuthURL:      cfg.AccountServiceURL,
+	}
+
+	oauthConfig := clientcredentials.Config{
+		ClientID:     svcConfig.ClientID,
+		ClientSecret: svcConfig.ClientSecret,
+		TokenURL:     svcConfig.AuthURL,
+	}
+
+	db, connection, err := storage.NewFromConfig(
+		cfg.Database,
+		events.Config{},
+		storage.NewEncrypter(cfg.Database.SecretKey),
+		logs.WithField("service", "storage"))
+	if err != nil {
+		logs.Error(err.Error())
+		return nil, nil, nil, nil, err
+	}
+
+	kcpK8sClient, err := getKcpClient()
+	if err != nil {
+		logs.Error(err.Error())
+		return nil, nil, nil, nil, err
+	}
+
+	svc := oauthConfig.Client(ctx)
+	return svc, db, connection, kcpK8sClient, nil
+}
+
+func fatalOnError(err error, log logrus.FieldLogger) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func getKcpClient() (client.Client, error) {
+	kcpK8sConfig, err := k8scfg.GetConfig()
+	mapper, err := apiutil.NewDiscoveryRESTMapper(kcpK8sConfig)
+	if err != nil {
+		err = wait.Poll(time.Second, time.Minute, func() (bool, error) {
+			mapper, err = apiutil.NewDiscoveryRESTMapper(kcpK8sConfig)
+			if err != nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("while waiting for client mapper: %w", err)
+		}
+	}
+	cli, err := client.New(kcpK8sConfig, client.Options{Mapper: mapper})
+	if err != nil {
+		return nil, fmt.Errorf("while creating a client: %w", err)
+	}
+	return cli, nil
+}
+
+func clusterOp(ctx context.Context, kcp client.Client, logs *logrus.Logger) (unstructured.UnstructuredList, error) {
+	gvk, err := k8s.GvkByName(k8s.KymaCr)
+	if err != nil {
+		logs.Errorf("error getting GVK %s", err)
+		return unstructured.UnstructuredList{}, nil
+	}
+
+	kymas := unstructured.UnstructuredList{}
+	kymas.SetGroupVersionKind(gvk)
+	err = kcp.List(ctx, &kymas)
+	if err != nil {
+		logs.Errorf("error listing kyma %s", err)
+		return unstructured.UnstructuredList{}, err
+	}
+	return kymas, nil
+}
+
+func dbOp(us *unstructured.Unstructured, db storage.BrokerStorage, logs *logrus.Logger) (internal.Instance, error) {
```

Comment:
It would be enough to pass only runtimeID instead of pointer to entire unstructured.Unstructured).